backend/controller/dal/dal.go

```diff
@@ -411,7 +411,7 @@ type IngressRoutingEntry struct {
 // previously created artefacts with it.
 //
 // If an existing deployment with identical artefacts exists, it is returned.
-func (d *DAL) CreateDeployment(ctx context.Context, language string, moduleSchema *schema.Module, artefacts []DeploymentArtefact, ingressRoutes []IngressRoutingEntry) (key model.DeploymentKey, err error) {
+func (d *DAL) CreateDeployment(ctx context.Context, language string, moduleSchema *schema.Module, artefacts []DeploymentArtefact, ingressRoutes []IngressRoutingEntry, cronJobs []CronJob, startTime time.Time) (key model.DeploymentKey, err error) {
```

Comment:
Is `startTime` ever going to be anything but `time.Now()`?

Reply:
Yes and no. `startTime` has to be used in the `CronJob.NextExecution` calculation, and we don't want startTime to progress after that.
Originally I had the db generating the StartTime value but ran into an edge case where the row was inserted as this due to the difference in time between the two "nows":
```
StartTime: 12:01.001
NextExecution: 12:01
State: idle
```
And thats an invalid state where the service will never schedule that job.

Reply:
That seems a bit fragile as it's up to the caller to ensure this implicit constraint is met. Can you do a check over the cron jobs NextExecution times, and create a timestamp that satisfies that constraint?

Reply:
That's a good idea

Reply:
nvm, that variable wasn't actually used anymore anyway 😅